### ISSUE TITLE
Introduce basic integration test of docker-registry on btp platform

### ISSUE THREAD

**Issue by kwiatekus** [kyma-project/docker-registry#127]
**Description**
Implement an e2e integration test for [docker-registry's basic flow](https://github.com/kyma-project/docker-registry/blob/main/docs/user/tutorials/01-10-use-registry.md).

**AC**
 - [x] The test should verify if a custom image could be pushed to docker-registry via external https endpoint (via docker CLI)
 - [x] The test should verify if a workload could be scheduled using image stored in internal docker registry and using (internal) image pull secret provided by the docker registry operator
 - [x] The test should be executed whenever a PR is merged into main branch ( part of "post-submit")
 - [ ] ... (and part of "release" workflows)
 - [x] The test should use [automation tooling](https://github.tools.sap/otters/kyma-automation-demo) designed to automate CI on top of btp platform resources (i.e bot user against a global account on canary landscape).
 - [x] The test should clean up btp resources via `btp remove accounts/subaccount --subbaccount {ID} --force`
 - [x] The test should ensure that all required binaries (kyma@v3, btp CLIs) are downloaded 
 - [x] the test job configuration should include: target btp landscape, target btp global account, bot user credentials (password read from sap vault), target custom IAS tenant, subaccount region, kyma plan and region. (see [config](https://github.tools.sap/otters/kyma-automation-demo/blob/main/env/.env-template-canary-azure) used in the poc)
 - [x] the test job should allow executing multiple runs in parallel - the resource names should be generated in a unique fashion (idea: incorporating hash of commit)

**Related**
Depends on https://github.com/orgs/kyma-project/projects/26/views/1?pane=issue&itemId=71063808
Depends on https://github.com/kyma-project/docker-registry/issues/98

**Reason**
We need an automated test that executes basic flow and protect us from regressions.

**Related issues**
https://github.com/kyma-project/kyma/issues/18555
